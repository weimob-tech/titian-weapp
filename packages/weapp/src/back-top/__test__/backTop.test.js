const { getComponent } = require('../../common/test/getComponent');

describe('test component', () => {
  it('should render success', () => {
    const comp = getComponent('back-top');
    expect(comp).toMatchSnapshot();
  });

  it('should use slot success', () => {
    const comp = getComponent('back-top', {
      useSlot: true
    });

    expect(comp).toMatchSnapshot();
  });

  it('should has text success', () => {
    const TEXT = '返回顶部';

    const comp = getComponent('back-top', {
      text: TEXT
    });
    expect(comp).toMatchSnapshot();
    expect(comp.querySelector('.titian-back-top-text').dom.innerHTML).toEqual(TEXT);
  });

  it('should trigger component attach lifeItems success', () => {
    const comp = getComponent('back-top');
    const onPageScroll = jest.fn();
    const currentPage = {
      onPageScroll
    };

    global.getCurrentPages = jest.fn(() => [currentPage]);
    comp.triggerLifeTime('attached');

    expect(global.getCurrentPages).toHaveBeenCalled();

    currentPage.onPageScroll({ scrollTop: 400 });

    expect(onPageScroll).toBeCalledTimes(1);
    expect(comp.instance.data.show).toBeTruthy();

    comp.setData({ visibilityHeight: 300 });
    expect(comp.instance.data.visibilityHeight).toBe(300);

    currentPage.onPageScroll({ scrollTop: 200 });
    expect(onPageScroll).toBeCalledTimes(2);
    expect(comp.instance.data.show).toBeFalsy();

    currentPage.onPageScroll({ scrollTop: 250 });
    expect(onPageScroll).toBeCalledTimes(3);
    expect(comp.instance.data.show).toBeFalsy();
  });

  it('should trigger click success', (done) => {
    const duration = 300;

    const pageScrollTo = jest.fn((e) => {
      expect(e.scrollTop).toBe(0);
      expect(e.duration).toBe(duration);
    });

    wx.pageScrollTo = pageScrollTo;

    const comp = getComponent('back-top', {
      duration
    });

    const onClick = jest.fn();
    comp.addEventListener('click', onClick);

    comp.querySelector('.titian-back-top').dispatchEvent('tap');

    Promise.resolve().then(() => {
      expect(onClick).toHaveBeenCalled();
      done();
    });
  });
});
