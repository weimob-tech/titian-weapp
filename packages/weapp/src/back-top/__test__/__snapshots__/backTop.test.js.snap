// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test component should has text success 1`] = `
<main>
  <wx-view
    class="ext-class  titian-back-top"
    style="display: none"
    bind:tap="onClick"
  >
    <ti-icon>
      <wx-view
        class="titian-icon-base titian-icon titian-icon-go-to-top ext-class"
        style="font-size:32rpx"
        bind:tap="onClick"
      />
    </ti-icon>
    <wx-text
      class="titian-back-top-text"
    >
      返回顶部
    </wx-text>
  </wx-view>
</main>
`;

exports[`test component should render success 1`] = `
<main>
  <wx-view
    class="ext-class  titian-back-top"
    style="display: none"
    bind:tap="onClick"
  >
    <ti-icon>
      <wx-view
        class="titian-icon-base titian-icon titian-icon-go-to-top ext-class"
        style="font-size:48rpx"
        bind:tap="onClick"
      />
    </ti-icon>
  </wx-view>
</main>
`;

exports[`test component should use slot success 1`] = `
<main>
  <wx-view
    class="ext-class  titian-back-top"
    style="display: none"
    bind:tap="onClick"
  />
</main>
`;
